### PR TITLE
DM-41142: Make metadetect a EUPS package

### PR DESCRIPTION
**Pull request by arunkannawadi**

### CODE REVIEW

File: ups/eupspkg.cfg.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+build() {
+   python setup.py pytest --addopts "metadetect/tests/ metadetect/lsst/tests/test_import.py"
```
Comment (arunkannawadi) [inline ups/eupspkg.cfg.sh:2]: Using `nosetests` does not obey the conditional skips, so switching to `pytest`.

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+name: lint
+
+on:
+  push:
+    branches:
+      - lsst-dev
+      - main
+      - master
```
Comment (TallJimbo) [inline .github/workflows/lint.yaml:8]: Do we really want to run our linters on anything other than `lsst-dev`?

File: .github/workflows/shear_test.yml [hunk 3]
```diff
@@ -11,7 +11,7 @@ jobs:
     name: shear-tests
     strategy:
       matrix:
-        pyver: ["3.10"]
+        pyver: ["3.10", "3.11"]
```
Comment (TallJimbo) [inline .github/workflows/shear_test.yml:14]: We can drop 3.9 and 3.10 here and in other workflows.

File: README.md [hunk 4]
```diff
@@ -1,5 +1,41 @@
...
+### Motivation
+
+We use a fork of this repository instead of declaring it as a dependency in `rubin-env` because the LSST-specific code in this repository uses some of the core packages of the LSST Science Pipelines themselves.
+Having a fork allows us keeps the dependency graph cleaner and simpler, while enabling the Science Pipelines
+team members to make any API changes in a consistent manner without breakage.
+Any significant algorithmic change is expected to happen in the upstream package and merged into this fork.
+
+The LSST-specific unit tests cannot be run on Jenkins because it has additional dependencies that are not available within `rubin-env` (e.g., `descwl-shear-sims`).
+However, these tests are the most relevant ones for the LSST organization as they would indicate any breakage.
+Therefore, we run these tests on GitHub Actions at least once weekly, using the latest weekly through `stackvana`.
```
Comment (TallJimbo) [inline README.md:24]: How do we get GHA to run tests weekly instead of on push?

**Review by TallJimbo — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (arunkannawadi) [reply README.md:24]: I've set up a cron job (on `schedule`) to run every Thursday night when the newer `stackvana` should be available if there were no build problems.

File: .github/workflows/lsst-tests-ngmix-upstream.yml [hunk 5]
```diff
@@ -3,15 +3,19 @@ name: lsst-tests-ngmix-upstream
 on:
   push:
     branches:
+      - lsst-dev
+      - main
       - master
   pull_request: null
+  schedule:
+    - cron: 0 23 * * 4
```
Comment (arunkannawadi) [inline .github/workflows/lsst-tests-ngmix-upstream.yml:11]: This workflow gets run every Thursday at 11pm UTC which is 3pm PT.

**Pull request merged**